test/e2e/resize_nodes.go

```diff
@@ -560,6 +566,93 @@ var _ = Describe("Nodes [Disruptive]", func() {
 					}
 				}
 			})
+
+			// What happens in this test:
+			// 	Network connectivity on a node is cut off for 120 seconds to simulate network partition
+			// Expect to observe:
+			// 1. Node is marked NotReady after timeout by nodecontroller (40seconds)
+			// 2. All pods on node are marked NotReady shortly after #1
+			// 3. Node and pods return to Ready after connectivivty recovers
```

Comment:
I believe the e2e framework waits for all node/pod to be ready before it starts and checks node readiness after each test. 

Before #17741, only kubelet manipulates pod status. Now nodecontroller can also mark pod notready. Hence kubelet cannot assume its pod status cache is always valid, this is fixed in #18410.  If just to test #18410, we still need to change pod status without telling kubelet and see if it could correct it. 

Basically this test wants to test #17741 and #18410